Ensure -webkit-overflow-scroll tests run in mobile mode

gsnedders opened the pull request:
#### c7f9e1da3dd819c25ec141ee483f1f62e63927a2
<pre>
Ensure -webkit-overflow-scroll tests run in mobile mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=243710">https://bugs.webkit.org/show_bug.cgi?id=243710</a>

Reviewed by Tim Horton.

iOS/iPadOS&apos;s &quot;desktop mode&quot; sets a policy that forces the
LegacyOverflowScrollingTouchEnabled setting to false; as a result, to
meaningfully test -webkit-overflow-scrolling: touch; we need to be in
mobile mode.

* LayoutTests/fast/scrolling/ios/overflow-scroll-inherited.html:
* LayoutTests/platform/ipad/TestExpectations:
* LayoutTests/platform/ipad/fast/scrolling/ios/overflow-scroll-inherited-expected.txt: Removed.
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-gain-scrolling-ancestor.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-in-fixed.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-lose-scrolling-ancestor.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/resources/doc-with-sticky.html:

Canonical link: <a href="https://commits.webkit.org/253361@main">https://commits.webkit.org/253361@main</a>
</pre>

hortont424 reviewed (approved):
r+ but Simon would like more words in the ChangeLog explaining what your change is (and also why you went that way instead of `useFlexibleViewport=true`, which is used in a lot of tests, but maybe is insufficient on iPad?) :)

gsnedders committed:
Ensure -webkit-overflow-scroll tests run in mobile mode
https://bugs.webkit.org/show_bug.cgi?id=243710

Reviewed by Tim Horton.

iOS/iPadOS's "desktop mode" sets a policy that forces the
LegacyOverflowScrollingTouchEnabled setting to false; as a result, to
meaningfully test -webkit-overflow-scrolling: touch; we need to be in
mobile mode.

* LayoutTests/fast/scrolling/ios/overflow-scroll-inherited.html:
* LayoutTests/platform/ipad/TestExpectations:
* LayoutTests/platform/ipad/fast/scrolling/ios/overflow-scroll-inherited-expected.txt: Removed.
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-gain-scrolling-ancestor.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-in-fixed.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame-lose-scrolling-ancestor.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/coordinated-frame.html:
* LayoutTests/scrollingcoordinator/scrolling-tree/resources/doc-with-sticky.html:

Canonical link: https://commits.webkit.org/253361@main
